internal/resources/gsi.go

```diff
@@ -582,12 +596,17 @@ func (g *GSI) executeGsiDdl(ctx context.Context, plan *providerschema.GsiDefinit
 		nil,
 	)
 	if err != nil {
+		if errors.Is(err, internalerrors.ErrGatewayTimeoutForIndexDDL) {
+			return internalerrors.ErrGatewayTimeoutForIndexDDL
+		}
```

Comment:
do you want to make this into a switch case?

Reply:
adding some comments on why we handle this error will be useful

Reply:
addressed in https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/357/commits/8aa7f12a65db2d271ef47f559ba4affe73243f5d